core/src/main/resources/org/apache/spark/ui/static/timeline-view.js

```diff
@@ -50,9 +50,18 @@ function drawApplicationTimeline(groupArray, eventObjArray, startTime, offset) {
        return "#job-" + jobId;
       };
 
+      var getPathForJobEntry = function(baseElem) {
+        var jobIdText = $($(baseElem).find(".application-timeline-content")[0]).text();
+        var jobId = jobIdText.match("\\(Job (\\d+)\\)$")[1];
+        return "job/?id=" + jobId;
```

Comment:
Did you confirm this change works with HistoryServer?
The base path for `AllJobsPage` is not `/` for HistoryServer so I'm afraid this code doesn't work with HistoryServer.